### PR TITLE
Format Histogram Tooltip to a readable form;

### PR DESCRIPTION
**Pull request by lon-io**
> Format Histogram Tooltip to a readable form;
> Add default for null localForage data;
> Separate logic for Top Domains from that for Histogram Data

### CODE REVIEW

File: src/app/utils.ts [hunk 1]
```diff
@@ -48,10 +48,70 @@ export const getFocusedTab = (): Promise<chrome.tabs.Tab> => {
...
   return domainData ? Object.values(domainData).reduce((acc, cur) => acc + cur.total_time, 0) : 0;
 }
 
+/**
+ * Pads a number with zeros
+ * @param val
+ * @param padding
+ * @returns string
+ */
+const padWithZeros = (val: number, padding?: number) => {
```
Comment (imolorhe) [inline src/app/utils.ts:65]: This is not needed. You can use the String.prototype.padStart() method.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

File: src/app/utils.ts [hunk 2]
```diff
@@ -48,10 +48,70 @@ export const getFocusedTab = (): Promise<chrome.tabs.Tab> => {
...
+  return `${pad}${val}`;
+}
+
+/**
+ * Pads with any arbitrary string(s)
+ * @param str
+ * @param pad
+ * @returns string
+ */
+const padWithString = (val: any, ...pad: any[]) => {
```
Comment (imolorhe) [inline src/app/utils.ts:79]: `String.prototype.padStart()`

File: src/app/utils.ts [hunk 3]
```diff
@@ -48,10 +48,70 @@ export const getFocusedTab = (): Promise<chrome.tabs.Tab> => {
...
+  // since the seconds value is a time difference)
+  const ss = date.getUTCSeconds();
+  const mm = date.getUTCMinutes();
+  const hh = date.getUTCHours();
+  const DD = date.getUTCDate() - leastDay;
+  const MM = date.getUTCMonth() + 1 - leastMonth;
+  const YY = date.getUTCFullYear() - leastYear;
+
+  let result = `${padWithZeros(ss)}s`;
+  if (mm > 0) result = padWithString(result, padWithZeros(mm), 'm');
```
Comment (imolorhe) [inline src/app/utils.ts:106]: 😄 I haven't seen this usage of pad before.

**Review by imolorhe — APPROVED** [review]

File: src/app/utils.ts (hunk 3 above)
Comment (lon-io) [reply src/app/utils.ts:106]: lol

**Pull request merged**